File: masar_chart/masar_chart/report/bin_report/bin_report.js
// Copyright (c) 2016, KCSC and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Bin Report"] = {
	"filters": [
		{
			"fieldname": "warehouse",
		  "label": __("Warehouse"),
		  "fieldtype": "Link",
			"options": "Warehouse",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "creation",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": 80,
			"reqd": 1,
			"default": dateutil.year_start()
		},
		{
			"fieldname": "item_code",
			"label": __("Item Code"),
			"fieldtype": "Link",
			"options": "Item",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "actual_qty",
			"label": __("Actual Quantity"),
			"fieldtype": "Data",
			"width": 100,
			"reqd": 0,
		},
		{
			"fieldname": "valuation_rate",
			"label": __("Valuation Rate"),
			"fieldtype": "Data",
			"width": 100,
			"reqd": 0,
		 }
	]
};

File: masar_chart/masar_chart/report/bin_report/bin_report.py
# Copyright (c) 2013, KCSC and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
from frappe import _
import frappe

def execute(filters=None):
	return get_columns(), get_data(filters)

def get_data(filters):
	print(f"\n\n\n\{filters}\n\n\n\n")
	# _creation = filters.get('creation'), filters.get('creation') #date range
	# condition
	# conditions = " AND 1=1 "
	# if(filters.get('warehouse')):conditions += f" AND warehouse='{filters.get('warehouse')}' "
	# if(filters.get('item_code')):conditions += f" AND item_code='{filters.get('item_code')}' "
	# print(f"\n\n\n\{conditions}\n\n\n\n")

	data = frappe.db.sql("""SELECT name, warehouse, item_code, actual_qty, stock_value, valuation_rate, creation FROM `tabBin`;""")
	return data

def get_columns():
	return [
	   "Name: Link/Bin:200",
	   "Warehouse:Data:150",
	   "Item Code:Data:150",
	   "Actual Quantity:Data:150",
	   "Stock Value:Data:150",
	   "Valuation Rate:Data:150",
	   "Creation:Data:150"
	]   
